src/myrtlespeech/model/rnn.py

```diff
@@ -135,6 +140,14 @@ def forward(
         if self.use_cuda:
             x = (x[0].cuda(), x[1].cuda())
 
+        if self.batch_norm is not None:
+            # Collapses input of dim T*N*H to (T*N)*H and gives it to a batch
```

Comment:
What about if the input to the rnn has dim=4?

Also - these comments are confusing in the batch_first=True case

Reply:
if the input has dim=4 then it would be necessary to use a batchnorm2d or change/collapse the input dimensions. I assumed that the input was always with dim=3 because it was explicitly said in the arg description of the forward function, but maybe it was a wrong assumption. Have you encountered situations in which the input dim is 4?

Regarding the comments, I have slightly changed them to clarify that it doesn't matter if batch_first is True or False because the first two dimensions are both collapsed into a single one. Let me know if they are still confusing.

Reply:
Ok you're absolutely right - dim=4 shouldn't work according to the docstring so the api shouldn't support this - my mistake. 

The batch_first comments are v clear now!